File: ussd-library/src/main/java/com/romellfudi/ussdlibrary/USSDApi.java
/*
 * Copyright (c) 2020. BoostTag E.I.R.L. Romell D.Z.
 * All rights reserved
 * porfile.romellfudi.com
 */

package com.romellfudi.ussdlibrary;

import java.util.HashMap;
import java.util.HashSet;

/**
 *
 * @author Romell Dominguez
 * @version 1.1.c 13/02/2018
 * @since 1.0.a
 */
public interface USSDApi {
    void send(String text, USSDController.CallbackMessage callbackMessage);
    void cancel();
    void callUSSDInvoke(String ussdPhoneNumber, HashMap<String,HashSet<String>> map,
                        USSDController.CallbackInvoke callbackInvoke);
    void callUSSDInvoke(String ussdPhoneNumber, int simSlot, HashMap<String,HashSet<String>> map,
                        USSDController.CallbackInvoke callbackInvoke);
    void callUSSDOverlayInvoke(String ussdPhoneNumber, HashMap<String,HashSet<String>> map,
                               USSDController.CallbackInvoke callbackInvoke);
    void callUSSDOverlayInvoke(String ussdPhoneNumber, int simSlot, HashMap<String,HashSet<String>> map,
                               USSDController.CallbackInvoke callbackInvoke);
}

File: ussd-library-kotlin/src/main/java/com/romellfudi/ussdlibrary/USSDApi.kt
/*
 * Copyright (c) 2020. BoostTag E.I.R.L. Romell D.Z.
 * All rights reserved
 * porfile.romellfudi.com
 */

/**
 * BoostTag E.I.R.L. All Copyright Reserved
 * www.boosttag.com
 */
package com.romellfudi.ussdlibrary

import android.content.Context
import java.util.*

/**
 *
 * @author Romell Dominguez
 * @version 1.1.i 2019/04/18
 * @since 1.1.i
 */
interface USSDApi {
    fun send(text: String, callbackMessage: (String) -> Unit)
    fun cancel()
    fun callUSSDInvoke(context: Context, ussdPhoneNumber: String,
                       map: HashMap<String, List<String>>,
                       callbackInvoke: USSDController.CallbackInvoke)

    fun callUSSDInvoke(context: Context, ussdPhoneNumber: String, simSlot: Int,
                       map: HashMap<String, List<String>>,
                       callbackInvoke: USSDController.CallbackInvoke)

    fun callUSSDOverlayInvoke(context: Context, ussdPhoneNumber: String,
                              map: HashMap<String, List<String>>,
                              callbackInvoke: USSDController.CallbackInvoke)

    fun callUSSDOverlayInvoke(context: Context, ussdPhoneNumber: String, simSlot: Int,
                              map: HashMap<String, List<String>>,
                              callbackInvoke: USSDController.CallbackInvoke)
    fun verifyAccessibilityAccess(context: Context): Boolean
    fun verifyOverLay(context: Context): Boolean
}
